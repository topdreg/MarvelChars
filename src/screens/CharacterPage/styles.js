import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    width: '100%',
    height: '100%',
    backgroundColor: 'white',
  },
  scrollView: {
    alignItems: 'center',
    height: '100%',
    width: '100%',
    paddingLeft: '10%',
    paddingRight: '10%',
  },
  name: {
    margin: 20,
    fontWeight: 'bold',
    fontSize: 18,
  },
  image: {
    height: 70,
    width: 70,
    marginBottom: 50,
  },
  descriptionContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    marginBottom: 30,
  },
  header: {
    marginBottom: 25,
    fontWeight: 'bold',
  },
  seriesContainer: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  series: {
    width: '100%',
  },
  seriesItem: {
    marginBottom: 10,
  },
});
