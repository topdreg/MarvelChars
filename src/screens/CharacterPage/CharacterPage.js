import React from 'react';
import {SafeAreaView, View, Text, Image, ScrollView} from 'react-native';
import FastImage from 'react-native-fast-image';
import {getRandom} from 'utilities/arrayOperations';
import BigTextDescription from 'screens/components/BigTextDescription';
import styles from './styles';

const CharacterPage = (props) => {
  const {characterData, imageUri} = props;
  const name = characterData?.name;
  const description = characterData?.description;
  const series = characterData?.series;
  // Get a randomized set of 7 elements from the series array, to be displayed in the Flatlist
  let selectedSeries = null;
  // First ensure that we have the series data
  if (series?.items && series.items.length > 0) {
    // Select up to 7 series items
    selectedSeries = getRandom(series.items, Math.min(7, series.items.length));
  }

  if (description === (null || '') && selectedSeries === null) {
    return (
      <BigTextDescription
        header="Come another time!"
        body="Data is currently not available for this character."
      />
    );
  }

  return (
    <SafeAreaView style={styles.container}>
      <ScrollView contentContainerStyle={styles.scrollView}>
        <Text style={styles.name}>{name}</Text>
        <FastImage style={styles.image} source={{uri: imageUri}} />
        {description !== (undefined || '') && (
          <View style={styles.descriptionContainer}>
            <Text style={styles.header}>Description</Text>
            <Text style={styles.description}>{description}</Text>
          </View>
        )}
        {selectedSeries !== null && (
          <View style={styles.seriesContainer}>
            <Text style={styles.header}>Series</Text>
            <View style={styles.series}>
              {selectedSeries.map((item, key) => {
                return (
                  <Text style={styles.seriesItem} key={key}>
                    {item.name}
                  </Text>
                );
              })}
            </View>
          </View>
        )}
      </ScrollView>
    </SafeAreaView>
  );
};

export default CharacterPage;
