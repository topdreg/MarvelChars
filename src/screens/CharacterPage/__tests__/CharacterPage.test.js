import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import CharacterPage from '../CharacterPage';

it('renders correctly with data', () => {
  const characterData = {
    name: 'A-Bomb',
    description: 'An awesome buddy!',
    series: {
      items: [
        {
          name: 'Comic 1!',
        },
        {
          name: 'Comic 2!',
        },
      ],
    },
  };
  const imageUri =
    'http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16.jpg';
  const tree = renderer
    .create(<CharacterPage characterData={characterData} imageUri={imageUri} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
