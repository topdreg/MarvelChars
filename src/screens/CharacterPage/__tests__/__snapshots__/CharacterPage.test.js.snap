// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "height": "100%",
        "paddingLeft": "10%",
        "paddingRight": "10%",
        "width": "100%",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
            "margin": 20,
          }
        }
      >
        A-Bomb
      </Text>
      <Image
        source={
          Object {
            "uri": "http://i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16.jpg",
          }
        }
        style={
          Object {
            "height": 70,
            "marginBottom": 50,
            "width": 70,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginBottom": 30,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
              "marginBottom": 25,
            }
          }
        >
          Description
        </Text>
        <Text>
          An awesome buddy!
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontWeight": "bold",
              "marginBottom": 25,
            }
          }
        >
          Series
        </Text>
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Comic 2!
          </Text>
          <Text
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            Comic 1!
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
